### PR TITLE
Add second template project

### PR DESCRIPTION
**Pull request by danieljurek**
The purpose of this project is to validate that we don't break the atomic vcpkg shipping scenario where multiple packages release in the same PR. Without this package we cannot meaningfully validate the atomic process in a way that doesn't cause packages (like `azure-core`) to be released (potentially multiple times). 

If we don't add this to the repo then it'll have to be added into nearly any branch which alters the release steps and then backed out. 

I'm open to changing the name from `azure-template-second` to something more like `azure-template-releasetester` or something which better describes its purpose. 

See vcpkg sample PR and CI at https://github.com/microsoft/vcpkg/pull/16399

### CODE REVIEW

**Comment by danieljurek** [comment]
@ahsonkhan -- you had some feedback back when I tried to sneak this in as part of another PR. 

@RickWinter -- your feedback here would also be appreciated given we're adding another package (intended to be used for testing only)

**Comment by ahsonkhan** [comment]
> I'm open to changing the name from `azure-template-second` to something more like `azure-template-releasetester` or something which better describes its purpose. 

Good idea. I am for it.

> If we don't add this to the repo then it'll have to be added into nearly any branch which alters the release steps and then backed out. 

How common would such release edits be, specific to the vcpkg release step? I am wondering how valuable this check will be going forward in actually guaranteeing successful releases of multiple packages, trading off against the extra "fluff" and noise to the repo its adding. Especially because we probably won't ship multiple packages in the same release pipeline step all that often. Maybe this could be an ad-hoc validation instead?

File: sdk/template/CMakeLists.txt [hunk 1]
```diff
@@ -9,3 +9,4 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS ON)
 
 add_subdirectory(azure-template)
+add_subdirectory(azure-template-second)
```
Comment (RickWinter) [inline sdk/template/CMakeLists.txt:12]: I don't have a strong preference on the name.  Maybe `engsys-testpkg` or `azure-engsys-testpkg`.  Something to make clear this package in about validating automation and should not be cloned as the starting point for client library authors.

```suggestion
add_subdirectory(azure-template-second)

```

File: sdk/template/azure-template-second/inc/azure/template/second/dll_import_export.hpp [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+/**
+ * @file
+ * @brief This file defines a macro for DLL export.
+ */
+
+// For explanation, see the comment in azure/core/dll_import_export.hpp
+
+#pragma once
+
+#if defined(AZ_TEMPLATE_DLL) || (0 /*@AZ_TEMPLATE_DLL_INSTALLED_AS_PACKAGE@*/)
```
Comment (RickWinter) [inline sdk/template/azure-template-second/inc/azure/template/second/dll_import_export.hpp:13]: There MACROS need to be named specific to the package name you choose.  i.e.
```suggestion
#if defined(AZ_TEMPLATE_SECOND_DLL) || (0 /*@AZ_TEMPLATE_SECOND_DLL_INSTALLED_AS_PACKAGE@*/)
```

File: sdk/template/azure-template-second/inc/azure/template/second/dll_import_export.hpp [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+#define AZ_TEMPLATE_BUILT_AS_DLL 1
+#else
+#define AZ_TEMPLATE_BUILT_AS_DLL 0
+#endif
+
+#if AZ_TEMPLATE_BUILT_AS_DLL
+#if defined(_MSC_VER)
+#if defined(AZ_TEMPLATE_BEING_BUILT)
+#define AZ_TEMPLATE_DLLEXPORT __declspec(dllexport)
```
Comment (RickWinter) [inline sdk/template/azure-template-second/inc/azure/template/second/dll_import_export.hpp:22]: ditto on all this and the unique name to the package

File: sdk/template/azure-template-second/inc/azure/template/second/template_client.hpp [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+#include <string>
+
+namespace Azure { namespace Template { namespace Second {
+
+  class TemplateClient {
+  public:
+    std::string const ClientVersion();
+  };
+
+}}} // namespace Azure::Template::Second
```
Comment (RickWinter) [inline sdk/template/azure-template-second/inc/azure/template/second/template_client.hpp:15]:
```suggestion
}}} // namespace Azure::Template::Second

```

File: sdk/template/azure-template-second/inc/azure/template/second/version.hpp [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+#pragma once
+
+#include "azure/template/second/dll_import_export.hpp"
+
+#include <string>
+
+#define AZURE_TEMPLATE_VERSION_MAJOR 1
+#define AZURE_TEMPLATE_VERSION_MINOR 0
+#define AZURE_TEMPLATE_VERSION_PATCH 0
+#define AZURE_TEMPLATE_VERSION_PRERELEASE "beta.30"
```
Comment (RickWinter) [inline sdk/template/azure-template-second/inc/azure/template/second/version.hpp:13]: This should start at beta.1

File: sdk/template/azure-template-second/vcpkg/CONTROL [hunk 6]
```diff
@@ -0,0 +1,10 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: MIT
+#
+Source: azure-template-second-cpp
+Version: @AZ_LIBRARY_VERSION@
+Build-Depends: azure-core-cpp
```
Comment (RickWinter) [inline sdk/template/azure-template-second/vcpkg/CONTROL:6]: Is this meant to only depend on core?  I thought template second was a dependency on template.

File: sdk/template/azure-template-second/vcpkg/CONTROL [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+Description: Microsoft Azure Template SDK for C++
+  This is a template library meant to illustrate initial client library development process for Azure SDK.
+  It is not meant to be published to vcpkg.
+Homepage: https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/template/azure-template
```
Comment (RickWinter) [inline sdk/template/azure-template-second/vcpkg/CONTROL:10]: Update link to the package name once you decide on the name
```suggestion
Homepage: https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/template/azure-template-second
```

**Review by RickWinter — CHANGES_REQUESTED** [review]

**Comment by danieljurek** [comment]
@ahsonkhan and I discussed this today and have come to the conclusion that this kind of testing could be a subset of tests we want to run periodically to ensure we aren't failing vcpkg CI. 

To that end it might be best to keep this in its own branch and use it to validate changes to vcpkg release until we can build the periodic vcpkg verification. Any such verification would have to use the exact same vcpkg release steps to ensure proper test coverage.

**Comment by RickWinter** [comment]
@danieljurek You mention you are going with a different approach.  Should this PR be closed?

**Comment by danieljurek** [comment]
Closing this PR as we are taking a different approach.

**Pull request closed without merging**